docs/Technical-Documentation/tech-memos/feature-toggle-system.md

```diff
@@ -0,0 +1,630 @@
+to ensure users are using the most up to date features. Because the flags are not expected to change that frequently, the Redux
+store is a good place to keep them while asynchronously checking for updates every so often. TDP already follows a similar pattern
+via it's "auth check" thunk. This technical memorandum recommends a similar approach for the feature flags.
+
+**Location**: `tdrs-frontend/src/actions/featureFlags.js` and `tdrs-frontend/src/reducers/featureFlags.js`
+
+**Action types**:
+- `FETCH_FEATURE_FLAGS` - Start fetching
+- `SET_FEATURE_FLAGS` - Store fetched flags
+- `SET_FEATURE_FLAGS_ERROR` - Store error
+- `CLEAR_FEATURE_FLAGS_ERROR` - Clear error
+- `RESET_FEATURE_FLAGS` - Clear all flags (on logout)
+
+**State shape**:
+```javascript
+{
+  featureFlags: {
+    flags: {
+      pia_submission: { enabled: true, config: {...} }
+    },
+    loading: false,
+    error: null,
+    lastFetched: 1700251234567,
+  }
+}
+```
+
+**Example Thunk**:
+```javascript
+export const fetchFeatureFlags = () => async (dispatch) => {
+  dispatch({ type: FETCH_FEATURE_FLAGS })
+  try {
+    const { data } = await axiosInstance.get(`${BACKEND_URL}/v1/features/`)
+    dispatch({ type: SET_FEATURE_FLAGS, payload: { flags: data } })
+  } catch (error) {
+    dispatch({ type: SET_FEATURE_FLAGS_ERROR, payload: { error } })
+  }
+}
+```
+
+### Auth Check Integration
+
+Because the feature flags are dependent on the auth state of the user, it makes sense to couple the execution of the `fetchAuth`
+with the handling of the feature flags. That is, when a user authenticates successfully, we should immediately fetch the feature
+flags and populate the redux store. Conversely, if the user does not successfully authenticate we should ensure the feature state
+is cleared. Because the auth check also runs via the `IdleTimer` we can guarantee a consistent and timely update to the feature
+flag state.
+
+**Location**: `tdrs-frontend/src/actions/auth.js`
+
+```javascript
+export const fetchAuth = () => async (dispatch) => {
+  dispatch({ type: FETCH_AUTH })
+  // ...Existing code
+  if (data?.user) {
+    // ...Existing code
+    // Update feature flags
+    dispatch(fetchFeatureFlags())
+  } else {
+    // ...Existing code
+    // Clear feature flags
+    dispatch(resetFeatureFlags())
+  }
+}
+```
+
+### Periodic Refresh Strategy
+
+In the event that the this pattern becomes more frequent, or the implementer decides not to couple the auth check with the
+feature flag update. This technical memorandum suggest a generic data component that will handle the interval based redux
+updates.
+
+**Example Component**:
+```javascript
+const ReduxRefresh = ({ thunk, intervalMs = 300000 }) => {
+  const dispatch = useDispatch()
+  const isAuthenticated = useSelector((state) => state.auth.authenticated)
+
+  useEffect(() => {
+    if (!isAuthenticated) return // Should probably clear redux state here
```

Comment:
> `Should probably clear the redux state here`

i'd assume we'd handle a clear state as all feature flags "off" (which makes sense generally for fail states), but it could be nice to have unauthenticated feature flag options too. Maybe that would be a future variation of `UserFeatureFlag` for anonymous users.